lib/parser.js

```diff
@@ -1,41 +1,184 @@
 'use strict';
 
-const HANDSHAKE_LENGTH = 36;
-const PARCEL_LENGTH = 16;
-const CHUNK_LENGTH = 12;
+const constants = {
+  PROTOCOL_VERSION: 1,
+  HANDSHAKE_LENGTH: 36,
+  PARCEL_LENGTH: 16,
+  CHUNK_LENGTH: 12,
+  STRUCT_PARCEL: 0,
+  STRUCT_CHUNK: 1,
+  PARCEL_PING: 0,
+  PARCEL_PONG: 1,
+  PARCEL_CALL: 2,
+  PARCEL_CALLBACK: 3,
+  PARCEL_EVENT: 4,
+  PARCEL_STREAM: 5
+};
 
-const parcel = (buffer) => {
+const readParcel = (
   // Read parcel from buffer
-  // Return parcel object
-  const rest = Buffer.from(buffer, 1, PARCEL_LENGTH);
-  return rest;
+  buffer // buffer
+  // Return: object
+) => {
+  const rest = buffer.slice(0, constants.PARCEL_LENGTH);
+  const parcel = {
+    structType: constants.STRUCT_PARCEL,
+    parcelId: rest.readInt32LE(1),
+    parcelType: rest.readInt8(5),
+    compression: rest.readInt8(6),
+    encoding: rest.readInt8(7),
+    length: parseInt(rest.slice(8).toString()),
+  };
+  return parcel;
+};
+
+const parcel = ({
+  // Encode parcel structure to Buffer
+  parcelId, // number, id of parcel: 4b
+  parcelType, // number, type of parcel
+  // 1b: ping = 0, pong = 1, call = 2, callback = 3, event = 4, stream = 5
+  compression, // number,
+  // 1b: no = 0, gzip = 1
+  encoding, // number, encoding type
+  // 1b: binary = 0, jstp = 1, json = 2, bson = 3, v8 = 4
+  length // string: 8b
```

Comment:
Why do length is a type of string?

Reply:
In `readParcel` you return parcel with length of type number. Same parameters types should stay consistent across all functions. One of possible solutions is to make length type of number and convert it inside function

Reply:
Apparently, Node supports 32-bit integers. Just a quick search gave me this: [buf.writeUInt32BE and buf.writeUInt32LE](https://nodejs.org/api/buffer.html#buffer_buf_writeuint32be_value_offset)

Reply:
We need to discuss this moment. Number of size 8 bytes can't be represented with string of size 8 bytes.

Reply:
JS currently doesn't support numbers bigger than `2^53 - 1`. Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER. It is possible with new `BigInt`, though. BTW, do we really need 64-bit length? Can you please give an example of use case?